appservice/src/createAppService/IAppServiceWizardContext.ts

```diff
@@ -93,4 +93,16 @@ export interface IAppServiceWizardContext extends IResourceGroupWizardContext, I
      * Indicates advanced creation should be used
      */
     advancedCreation?: boolean;
+
+    customLocation?: CustomLocation;
+}
+
+export type CustomLocation = {
```

Comment:
Feels like this should extend `Location` and just add the `kubeEnvironment` property.

```suggestion
export type CustomLocation = Location & {
    kubeEnvironment: {
        name: string;
        id: string;
        location: string;
    }
}
```